### PR TITLE
[cmake] add -headerpad_max_install_names when building dylibs

### PR DESCRIPTION
**Pull request by justice-adams-apple**
add ` -headerpad_max_install_names` linker flag when appropriate

### CODE REVIEW

**Review by compnerd — APPROVED** [review]
gross

File: Runtimes/Core/core/CMakeLists.txt [hunk 1]
```diff
@@ -265,6 +265,12 @@ if(SwiftCore_ENABLE_VECTOR_TYPES)
     "${CMAKE_CURRENT_BINARY_DIR}/SIMDVectorTypes.swift")
 endif()
 
+if(APPLE AND BUILD_SHARED_LIBS)
+  target_link_options(swiftCore PRIVATE
+    "SHELL:-Xlinker -headerpad_max_install_names"
+  )
```
Comment (etcwilde) [inline Runtimes/Core/core/CMakeLists.txt:271]: To be consistent with the rest of this file, can we keep the closing `)` on the previous line? Thanks

**Comment by justice-adams-apple** [comment]
@swift-ci please smoke test

**Review by etcwilde — APPROVED** [review]

**Comment by justice-adams-apple** [comment]
@swift-ci please smoke test

**Pull request merged**